src/components/vuestic-components/va-divider/VaDivider.vue

```diff
@@ -0,0 +1,146 @@
+<template>
+  <div :class="classComputed">
+    <div
+      v-if="hasSlot && !vertical"
+      :class="slotClassComputed"
+    >
+      <slot />
+    </div>
+  </div>
+</template>
+
+<script>
+const prefixCls = 'va-divider'
+
+export default {
+  name: 'VaDivider',
+  props: {
+    vertical: {
+      type: Boolean,
+      default: false,
+    },
+    dashed: {
+      type: Boolean,
+      default: false,
+    },
+    inset: {
+      type: Boolean,
+      default: false,
+    },
+    orientation: {
+      type: String,
+      validator (value) {
+        return ['left', 'right', 'center'].indexOf(value) !== -1
+      },
+      default: 'center',
+    },
+  },
+  computed: {
+    hasSlot () {
+      return !!this.$slots.default
+    },
+    classComputed () {
+      return [
+        `${prefixCls}`,
+        {
+          [`${prefixCls}--vertical`]: this.vertical,
+          [`${prefixCls}--inset`]: this.inset && !this.vertical,
+          [`${prefixCls}--${this.orientation}`]: this.orientation && !this.vertical,
+          [`${prefixCls}--dashed`]: this.dashed,
+        },
+      ]
+    },
+    slotClassComputed () {
+      return `${prefixCls}__text`
+    },
+  },
+}
+</script>
+
+<style lang="scss">
+  @import "../../vuestic-sass/resources/resources";
+
+  $va-divider-line-width: 1px;
+  $va-divider-inset-margin: 5rem;
+
+  // This is a bit tricky one in case of a horizontal divider,
+  // but in fact, it adds gaps at the top and bottom of the divider line
+  // In case of a vertical divider, it adds gaps at the right and left as expected
+  $va-divider-margin: 0.5rem;
+
+  // This one adds offsets at the right and left for the given text orientation
+  // only for horizontal divider
+  $va-divider-text-horizontal-offset: 1.25rem;
+
+  $va-divider-text-font-size: 0.875rem;
+
+  .va-divider {
+    display: flex;
+    margin-top: $va-divider-margin;
+
+    & + * {
+      margin-top: -$va-divider-margin;
+    }
+
+    &--vertical + * {
+      margin-top: unset;
+    }
+
+    &--vertical {
+      margin: 0 $va-divider-margin;
+      border-top: 0;
+      border-right-width: $va-divider-line-width;
+      border-right-style: solid;
+      border-right-color: $separator-color;
+
+      &.va-divider--dashed {
+        border-right-style: dashed;
+      }
+    }
+
+    &--inset {
+      margin-left: $va-divider-inset-margin;
```

Comment:
Current implementation works properly only for list items (which I can't say is hugely important case).
Please check vuetify implementation without the list.